processing/src/main/java/io/druid/query/aggregation/first/StringFirstAggregatorFactory.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.first;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Longs;
+import io.druid.query.aggregation.AggregateCombiner;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.AggregatorUtil;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.query.aggregation.SerializablePairLongString;
+import io.druid.query.cache.CacheKeyBuilder;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.column.Column;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@JsonTypeName("stringFirst")
+public class StringFirstAggregatorFactory extends AggregatorFactory
+{
+  public static final int DEFAULT_MAX_STRING_SIZE = 1024;
+
+  public static final Comparator TIME_COMPARATOR = (o1, o2) -> Longs.compare(
+      ((SerializablePairLongString) o1).lhs,
+      ((SerializablePairLongString) o2).lhs
+  );
+
+  public static final Comparator VALUE_COMPARATOR = (o1, o2) -> {
+    String s1 = null;
+    String s2 = null;
+
+    if (o1 != null) {
+      s1 = ((SerializablePairLongString) o1).rhs;
+    }
+    if (o2 != null) {
+      s2 = ((SerializablePairLongString) o2).rhs;
+    }
+
+    return Objects.equals(s1, s2) ? 1 : 0;
```

Comment:
I still don't understand this line. Would you elaborate more on this?

Reply:
I'm thinking that maybe this is not correct and the correct comparator logic is this:

https://github.com/wizzie-io/druid/blob/9829263e7344b551904ec9eba5aab732d121e311/processing/src/main/java/io/druid/query/aggregation/SerializablePairLongStringSerde.java#L86

What do you think??